app/src/org/commcare/CommCareApp.java

```diff
@@ -299,28 +300,61 @@ public AndroidCommCarePlatform getCommCarePlatform() {
         return platform;
     }
 
-    public boolean hasVisibleTrainingMenu() {
+    public boolean hasVisibleTrainingContent() {
         // This is the same eval context that would be used to evaluate the relevancy conditions of
         // these menus when they're actually loaded
         EvaluationContext ec =
                 CommCareApplication.instance().getCurrentSessionWrapper().getEvaluationContext(Menu.TRAINING_MENU_ROOT);
-
         for (Suite s : platform.getInstalledSuites()) {
-            List<Menu> trainingMenus = s.getMenusWithRoot(Menu.TRAINING_MENU_ROOT);
-            if (trainingMenus != null) {
-                for (Menu m : trainingMenus) {
-                    try {
-                        if (m.getMenuRelevance() == null ||
-                                FunctionUtils.toBoolean(m.getMenuRelevance().eval(ec))) {
-                            return true;
-                        }
-                    } catch (XPathSyntaxException e) {
-                        // Now is the wrong time to show the user an error about this since they
-                        // haven't actually navigated to the menu. To be safe, just assume that this
-                        // menu is visible, and then if they navigate to it they'll see the XPath error there
+            if (visibleMenusInTrainingRoot(s, ec) || visibleEntriesInTrainingRoot(s, ec)) {

```

Comment:
Both these methods can be static. We might also want to extract them out to a helper class.

Reply:
@shubham1g5 Thanks! I made them static but after looking through our existing helper classes for a bit, i didn't find any that seemed fitting, and I think it's a little overkill to create a new helper class for just 2 methods.